src/main/java/org/sonarsource/sonarlint/ls/SonarLintExtendedLanguageClient.java

```diff
@@ -111,12 +111,24 @@ public String getConfigurationScopeId() {
     }
   }
 
-  record ShowFixSuggestionParams(String suggestionId, List<ChangesDto> textEdits, String fileUri) {
+  record ShowFixSuggestionParams(String suggestionId, List<ChangesDto> textEdits, String fileUri, boolean isLocal) {
   }
 
   @JsonNotification("sonarlint/showFixSuggestion")
   void showFixSuggestion(ShowFixSuggestionParams params);
 
+  record StartProgressNotificationParams(String taskId, String message) {
+  }
+
+  record EndProgressNotificationParams(String taskId) {
+  }
+
+  @JsonNotification("sonarlint/startProgressNotification")
+  void startProgressNotification(StartProgressNotificationParams params);
+
+  @JsonNotification("sonarlint/endProgressNotification")
+  void endProgressNotification(EndProgressNotificationParams params);
+
```

Comment:
Why not use the LSP built-in progress support?

Reply:
Because I want to show a pop-up progress notification, and the built-in progress stuff shows the progress in the status bar, which is very hidden. It's possible I missed something
![Screenshot 2025-04-07 at 14 51 13](https://github.com/user-attachments/assets/058a9669-8d2f-43f2-b5e6-d08807f1d7ea)